### PR TITLE
Raw date time strings are now scraped and presented as received

### PR DESCRIPTION
**Pull request by dfsm**
Raw strings representing dates and times are now captured as-is during scraping. These values are now displayed in most places of the app. This includes regdates, postdates, sentDates, etc.

(Except for the message list view, which uses a date formatter appropriate for a message list.)

Also, executive decision has been made to drop the "Joined" text preceding user regdates in Post.html.stencil. My case: The site doesn't do it

### CODE REVIEW

File: App/Data Sources/MessageListDataSource.swift [hunk 1]
```diff
@@ -96,6 +96,10 @@ extension MessageListDataSource: UITableViewDataSource {
     private func viewModelForMessage(at indexPath: IndexPath) -> MessageListCell.ViewModel {
         let message = self.message(at: indexPath)
         let theme = Theme.defaultTheme()
+          
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "d MMM"
```
Comment (nolanw) [inline App/Data Sources/MessageListDataSource.swift:101]: Let's move this formatter and its configuration to an instance property on `MessageListDataSource`. It's surprisingly expensive to change the format.

File: AwfulCore/Sources/AwfulCore/Model/Awful.xcdatamodeld/.xccurrentversion [hunk 2]
```diff
@@ -3,6 +3,6 @@
 <plist version="1.0">
 <dict>
 	<key>_XCCurrentVersionName</key>
-	<string>Awful 6.2.xcdatamodel</string>
+	<string>Awful 6. 3.xcdatamodel</string>
```
Comment (nolanw) [inline AwfulCore/Sources/AwfulCore/Model/Awful.xcdatamodeld/.xccurrentversion:6]: Weird version name, with the space before `3` :) Purely cosmetic but kinda weird.

File: AwfulCore/Sources/AwfulCore/Model/Post.swift [hunk 3]
```diff
@@ -27,6 +27,9 @@ public class Post: AwfulManagedObject, Managed {
     /// When the post appeared.
     @NSManaged public var postDate: Date?
     
+    /// Raw post date straight from the forums
+    @NSManaged public var postDateRaw: String
```
Comment (nolanw) [inline AwfulCore/Sources/AwfulCore/Model/Post.swift:31]: This is marked `optional` in the `.xcdatamodel` so it should be `String?` here, I think?

Alternately, ok to default to empty string in the model and leave this as-is. But one or the other.

**Review by nolanw — CHANGES_REQUESTED** [review]
We should pick one or the other for `postDateRaw` optionality.

Fair enough on dropping "Joined". I think I put it in because having two otherwise unlabelled dates in each post might be confusing. Probably obvious though.

File: App/Data Sources/MessageListDataSource.swift (hunk 1 above)
Comment (dfsm) [reply App/Data Sources/MessageListDataSource.swift]: I've addressed this by moving it into a pre-existing sentdateformatter variable

File: AwfulCore/Sources/AwfulCore/Model/Awful.xcdatamodeld/.xccurrentversion (hunk 2 above)
Comment (dfsm) [reply AwfulCore/Sources/AwfulCore/Model/Awful.xcdatamodeld/.xccurrentversion]: This should be resolved! I published a new model version with the correct naming and deleted the borked one. I've cloned this repo after pushing that change and all seems to be working OK

File: AwfulCore/Sources/AwfulCore/Model/Post.swift (hunk 3 above)
Comment (dfsm) [reply AwfulCore/Sources/AwfulCore/Model/Post.swift]: Made it optional!

**Comment by nolanw** [comment]
Excellent! ty

**Pull request merged**